flocker/common/version.py

```diff
@@ -158,8 +160,74 @@ def is_pre_release(version):
     :param bytes version: A version of flocker.
     :return bool: Whether the version is a pre-release.
     """
-    parsed_version = parse_version(version)
+    parsed_version = _parse_version(version)
     return (parsed_version.pre_release is not None
             and parsed_version.weekly_release is None
             and parsed_version.commit_count is None
             and parsed_version.dirty is None)
+
+
+class RPMVersion(PRecord):
+    """
+    An RPM compatible version and a release version.
+    See: http://fedoraproject.org/wiki/Packaging:NamingGuidelines#Pre-Release_packages  # noqa
+
+    :ivar bytes version: An RPM compatible version.
+    :ivar bytes release: An RPM compatible release version.
+    """
+    version = field(mandatory=True)
+    release = field(mandatory=True)
+
+
+def make_rpm_version(flocker_version):
+    """
+    Parse the Flocker version generated by versioneer into a
+    :class:`RPMVersion`.
+
+    :param flocker_version: The versioneer style Flocker version string.
+    :return: An ``RPMVersion``.
+    """
+    # E.g. 0.1.2-69-gd2ff20c-dirty
+    # tag+distance+shortid+dirty
+    parts = flocker_version.split('-')
```

Comment:
I think it should be fairly easy to convert this parsing to use `_parse_version` above, and then formating the parts. If it is straightforward, do this now, otherwise, file an issue.

A related suggestion is to make this a method (`os_version`?) on `FlockerVersion`. (Again, this may be a better as a follow-up issue).